sceptre/helpers.py

```diff
@@ -117,3 +119,21 @@ def null_context():
     available in py3.6, so providing it here instead.
     """
     yield
+
+
+def extract_datetime_from_aws_response_headers(boto_response):
+    """Returns a datetime.datetime extracted from the response metadata in a
+    boto response or None if it's unable to find or parse one.
+    :param boto_response: A dictionary returned from a boto client call
+    :type boto_response: dict
+    :returns a datetime.datetime or None
+    """
```

Comment:
Now that we're in the land of Python 3, it doesn't make any sense to indicate types in docstrings. Just use type annotations! They're actually readable by code and work better for many reasons. 

```suggestion
# you'll need to import Optional and datetime.datetime for these annotations to work
def extract_datetime_from_aws_response_headers(boto_response: dict) -> Optional[datetime]:
    """Returns a datetime.datetime extracted from the response metadata in a
    boto response or None if it's unable to find or parse one.
    :param boto_response: A dictionary returned from a boto client call
    :returns a datetime.datetime or None
    """
```

Reply:
I'd wondered about that but went for consistency with the other functions. Will do.